tsdate/evaluation.py

```diff
@@ -252,3 +252,107 @@ def match_node_ages(ts, other):
     matched_time[matched_span == 0] = np.nan
 
     return matched_time, matched_span, best_match
+
+
+def node_spans(ts):
+    """
+    Returns the array of "node spans", i.e., the `j`th entry gives
+    the total span over which node `j` is in the tree (i.e., does
+    not have 'missing data' there).
+    """
+    child_spans = np.bincount(
+        ts.edges_child,
+        weights=ts.edges_right - ts.edges_left,
+        minlength=ts.num_nodes,
+    )
+    for t in ts.trees():
+        span = t.span
+        for r in t.roots:
+            # do this check to exempt 'missing data'
+            if t.num_children(r) > 0:
+                child_spans[r] += span
+    return child_spans
+
+
+def tree_discrepancy(ts, other):
+    """
+    For two tree sequences `ts` and `other`,
+    this method `tree_discrepancy` returns two values:
+    1. the sum across the nodes of `ts`
+    of the length of the discrepancy in span between the node
+    and its best match node in `other`
+    weighted by difference in time.
+    2. The root mean squared difference
+    between the times in the nodes in `ts`
+    and times of their best matching nodes in `other`
+    with the average weighted by the span in `ts`.
+
+    This is done as follows:
+
+    For each node in `ts` the best matching node(s) from `other`
+    has the longest matching span using `shared_node_spans`.
+    If either tree sequence contains unary nodes there may be
+    multiple matches with the same longest shared span
+    for a single node.
+    In this case, the best match is the node closest in time.
```

Comment:
```suggestion
    If there are multiple matches with the same longest shared span
    for a single node, the best match is the match that is closest in time.
```
There could be multiple matches without unary nodes if there are non-sample leaves.